src/python_tools/tftp_transfer/host.py

```diff
@@ -0,0 +1,100 @@
+import tftpy, hashlib, os, argparse, pathlib, shutil
+
+TEMP = os.path.join(os.getcwd(), ".temp")
+
+
+def verify_checksum(file_path: str, checksum: str) -> bool:
+    """
+    Compare the SHA-256 checksum of a given file and verify that it is the same as a given checksum.
+    
+    Any file-reading errors will return false.
+    
+    :param file_path: the path to a file
+    :param checksum: the target SHA-256 checksum to compare against
+    :return true if both checksums are correct, false otherwise
+    """
+    try:
+        file_hash = encode_sha256_checksum(file_path)
+    except Exception:
+        return False
```

Comment:
this can hide any unintended exceptions that occur, its okay to catchall here, but we should log the exception 

id recommend doing 
~~~
except Exception as exc:
    logger.exception("Unexpected error when computing checksum")
    return False
~~~